taprpc/taprootassets.proto

```diff
@@ -628,9 +645,10 @@ message AssetTransfer {
     // Describes the set of newly created asset outputs.
     repeated TransferOutput outputs = 6;
 
-    // The block hash the contains the anchor transaction above. If unset, the
-    // anchor transaction is unconfirmed.
-    bytes anchor_tx_block_hash = 7;
+    // The block hash of the blockchain block that contains the anchor
+    // transaction. If this value is unset, the anchor transaction is
+    // unconfirmed.
+    BlockHash anchor_tx_block_hash = 7;
```

Comment:
If we want consistency across the project, I think we should use this for all occurrences of a block or transaction hash. But that would also mean a breaking change for developers...

I know you weigh the urge to "make things nice and clean and proper" quite high.
My tendency in general is more along the lines of "never break APIs" and "at least be consistent across APIs, even if it doesn't feel super clean"...
So my _slight_ tendency would be to just go with two fields here as well and not introduce a new message. Because that would just mean yet another pattern for developers to get used to (even if I agree that it's probably how things should've been from the start). And since we don't normally change any of our APIs (at least definitely in `lnd` that's the policy), we couldn't upgrade all other APIs to the new pattern...

Reply:
The two fields still exist though, so I don't see this as breaking the two fields pattern. It just wraps them in a single message. Gives us a chance to document a bit better, and re-use for un/marshalling, and so on. I don't think we should break any existing fields (the field in the PR is an exception because it hasn't been released yet).

Reply:
Yeah, okay, I guess you're right. Let's do it for this field and new fields.